test/unit/alignment/multiple/align_multiple_test.cpp

```diff
@@ -84,3 +111,146 @@ TEST(align_multiple_test, the_first_aminoacid_test)
+                                                                                    seqan3::gap_open_score{-1}}};
+    std::vector<seqan3::rna4_vector> input{"UUUACCCGAG"_rna4, "UUCCCGCG"_rna4, "UUUACGU"_rna4};
+    seqan3::gap g{};
+    //with go = -1, g = -1
+    auto seqan3_interface_result = seqan3::align_multiple(input, cfg);
+
+    //alignment with seqan2 --------------------------------------------------------------------------------------------
+    using alphabet_type = seqan::Rna;
+
+    using seqan3_alphabet_type = seqan3::rna4;
+    using sequence_type = seqan::String<alphabet_type>;
+    using graph_type = seqan::Graph<seqan::Alignment<seqan::StringSet<sequence_type, seqan::Dependent<>>,
+                                                     void,
+                                                     seqan::WithoutEdgeId>>;
+    using gapped_alphabet_type = seqan3::gapped<seqan3_alphabet_type>;
+
+    seqan::MsaOptions<seqan::Rna, seqan::Score<int>> msaOpt{};
+    msaOpt.build = 0; // NJ for building the guide tree.
+    seqan::appendValue(msaOpt.method, 0); // global alignment
+    seqan::appendValue(msaOpt.method, 1); // local alignment
+    msaOpt.pairwiseAlignmentMethod = 1; // unbanded
+    seqan::Score<int> scMat;
+    msaOpt.sc = scMat;
+    msaOpt.sc.data_gap_open = -2; // seqan2 does not add a gap score for the first gap character but just the go score.
+                                  // seqan3 does add the gap score for the first gap character, so we add -1 to the
+                                  // seqan2 go score.
+    msaOpt.sc.data_gap_extend = -1;
+    msaOpt.sc.data_match = 5;       // tcoffee app default
+    msaOpt.sc.data_mismatch = -4;   // tcoffee app default
+    // fill seqan2 data storage
+    seqan::StringSet<sequence_type, seqan::Owner<>> sequenceSet;
+    seqan::appendValue(sequenceSet, "UUUACCCGAG");
+    seqan::appendValue(sequenceSet, "UUCCCGCG");
+    seqan::appendValue(sequenceSet, "UUUACGU");
+    seqan::StringSet<seqan::String<char>> sequenceNames;
+    seqan::appendValue(sequenceNames, "1");
+    seqan::appendValue(sequenceNames, "2");
+    seqan::appendValue(sequenceNames, "3");
+
+    // Alignment of the sequences
+    graph_type gAlign;
+
+    // MSA
+    seqan::globalMsaAlignment(gAlign, sequenceSet, sequenceNames, msaOpt);
+
+    std::string mat;
+    seqan::convertAlignment(gAlign, mat);
+
+    using gapped_alphabet_type = seqan3::gapped<seqan3_alphabet_type>;
+    std::vector<std::vector<gapped_alphabet_type>> seqan2_msa_result{seqan::length(seqan::stringSet(gAlign))};
+    size_t ali_len = mat.size() / seqan2_msa_result.size();
+
+    auto iter = seqan2_msa_result.begin();
+    for (auto && gapped_seq : mat | seqan3::views::char_to<gapped_alphabet_type> | seqan3::views::chunk(ali_len))
+    {
+        iter->reserve(ali_len);
+        std::ranges::copy(gapped_seq, std::cpp20::back_inserter(*iter++));
+    }
+
+    EXPECT_RANGE_EQ(seqan2_msa_result, seqan3_interface_result);
```

Comment:
The conversion to the seqan3 data structure is quite complex and actually not needed here in order to simply compare the result alignment. I suggest removing this part and using:
```cpp
EXPECT_RANGE_EQ(mat, seqan3_interface_result | seqan3::views::to_char | seqan3::views::join);
```
It transforms the `vector<vector<gapped<rna4>>>` of the seqan3 output to a single char sequence and directly compares it with the seqan2 string representation.

Reply:
Thanks, this looks cleaner and is way shorter :)